File: packages/xyd-opensdk-cli/src/cli/write-report.ts
import type { WriteProjectResult } from '@xyd-js/opensdk-framework';

/**
 * Report a writeProject outcome: file count + the never-silent buckets (merges,
 * conflicts, kept files). Shared by the SDK emit path and the CLI-target path.
 */
export function reportWriteResult(fileCount: number, output: string, result: WriteProjectResult): void {
  console.log(`Generated ${fileCount} files in ${output}`);
  // Merge outcomes (--merge): clean 3-way merges + conflicts to resolve.
  for (const rel of result.merged) {
    console.log(`  ✓ merged your edits into ${rel}`);
  }
  for (const rel of result.mergeConflicts) {
    console.warn(`  ⚠ merge conflict in ${rel} — resolve the <<<<<<< markers`);
  }
  // .sdkignore conflicts + kept-modified orphans: never silently overwritten/lost.
  for (const rel of result.conflicts) {
    console.warn(`  ⚠ .sdkignore: kept your ${rel} — generated output differs (not overwritten)`);
  }
  for (const rel of result.keptModified) {
    console.warn(`  ⚠ kept locally-modified ${rel} — no longer generated, not pruned`);
  }
}

File: crates/xyd_opensdk_cli/src/write_report.rs
//! `src/cli/write-report.ts` — report a `write_project` outcome: the file count
//! plus the never-silent buckets (merges, conflicts, kept files). Shared by the
//! SDK emit path and the CLI-target path.

use xyd_opensdk_framework::WriteProjectResult;

pub fn report_write_result(file_count: usize, output: &str, result: &WriteProjectResult) {
    println!("Generated {file_count} files in {output}");
    // Merge outcomes (--merge): clean 3-way merges + conflicts to resolve.
    for rel in &result.merged {
        println!("  ✓ merged your edits into {rel}");
    }
    for rel in &result.merge_conflicts {
        eprintln!("  ⚠ merge conflict in {rel} — resolve the <<<<<<< markers");
    }
    // .sdkignore conflicts + kept-modified orphans: never silently overwritten.
    for rel in &result.conflicts {
        eprintln!("  ⚠ .sdkignore: kept your {rel} — generated output differs (not overwritten)");
    }
    for rel in &result.kept_modified {
        eprintln!("  ⚠ kept locally-modified {rel} — no longer generated, not pruned");
    }
}
